src/rgw/rgw_rest_swift.cc

```diff
@@ -1325,9 +1326,143 @@ void RGWBulkDelete_ObjStore_SWIFT::send_response()
   rgw_flush_formatter_and_reset(s, s->formatter);
 }
 
+vector<pair<string, RGWInfo_ObjStore_SWIFT::info>> RGWInfo_ObjStore_SWIFT::swift_info =
+  {
+    {"bulk_delete", {false, NULL}},
+    {"container_quotas", {false, NULL}},
+    {"swift", {false, RGWInfo_ObjStore_SWIFT::list_swift_data}},
+    {"tempurl", { false, RGWInfo_ObjStore_SWIFT::list_tempurl_data}},
+    {"slo", {false, RGWInfo_ObjStore_SWIFT::list_slo_data}},
+    {"account_quotas", {false, NULL}},
+    {"staticweb", {false, NULL}},
+    {"tempauth", {false, NULL}},
+  };
+
+void RGWInfo_ObjStore_SWIFT::execute()
+{
+  bool is_admin_info_enabled = false;
+
+  const string& swiftinfo_sig = s->info.args.get("swiftinfo_sig");
+  const string& swiftinfo_expires = s->info.args.get("swiftinfo_expires");
+
+  if (!swiftinfo_sig.empty() && !swiftinfo_expires.empty() && !is_expired(swiftinfo_expires))
+  {
+    is_admin_info_enabled = true;
+  }
+
+  s->formatter->open_object_section("info");
+
+  for (auto it = swift_info.begin(); it != swift_info.end(); it++)
```

Comment:
It would be nice to see the ranged `for` variant here: `for (const auto& pair : swift_info) { ... }`.